cmd/tsidp/tsidp_test.go

```diff
@@ -824,3 +826,159 @@ func TestExtraUserInfo(t *testing.T) {
 		})
 	}
 }
+
+func TestFunnelClientsPersistence(t *testing.T) {
+	log.SetOutput(io.Discard) // suppress log output during tests
```

Comment:
this has side effects that survive when the test is done

Can you instead not use a global logger in the code and support an optional one on idpServer?